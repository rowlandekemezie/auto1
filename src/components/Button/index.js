import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Button = styled("button")`
  width: 128px;
  height: 32px;
  background-color: ${({ theme }) => theme.LIGHT_ORANGE};
  color: white;
  border: none;
  border-radius: 3px;
  transition: all 0.3s ease;
  cursor: pointer;
  outline: none;

  &:active,
  &:focus,
  &:hover {
    background-color: ${({ theme }) => theme.ORANGE};
  }
`;

export default ({ label, ...props }) => <Button {...props}>{label}</Button>;

Button.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func
};
