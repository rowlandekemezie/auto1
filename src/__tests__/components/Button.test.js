import React from "react";
import { shallow, mount } from "enzyme";
import Button from "../../components/Button";

describe("<Button />", () => {
  it("should render component", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.length).toBe(1);
    expect(wrapper.find("button")).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render component with props", () => {
    const props = { label: "Title", onClick: jest.fn() };
    const wrapper = mount(<Button {...props} />);
    expect(wrapper.props().label).toBe(props.label);
    expect(wrapper.props().onClick).toBe(props.onClick);
  });
});
