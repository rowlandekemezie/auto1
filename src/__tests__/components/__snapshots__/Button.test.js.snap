// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: 128px;
  height: 32px;
  background-color: ",
          [Function],
          ";
  color: white;
  border: none;
  border-radius: 3px;
  transition: all 0.3s ease;
  cursor: pointer;
  outline: none;

  &:active,
  &:focus,
  &:hover {
    background-color: ",
          [Function],
          ";
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "label": [Function],
        "onClick": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  width: 128px;
  height: 32px;
  background-color: ",
            [Function],
            ";
  color: white;
  border: none;
  border-radius: 3px;
  transition: all 0.3s ease;
  cursor: pointer;
  outline: none;

  &:active,
  &:focus,
  &:hover {
    background-color: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "label": [Function],
          "onClick": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
